api/src/core/generation/BoardGenerator.ts

```diff
@@ -115,8 +115,8 @@ export class BoardGenerator {
             }
             goal.categories.forEach((cat) => {
                 const prev =
-                    this.goalsByCategory[this.categoriesByName[cat].id] ?? [];
-                this.goalsByCategory[this.categoriesByName[cat].id] = [
+                    this.goalsByCategoryId[this.categoriesByName[cat.name].id] ?? [];
```

Comment:
Same as earlier, we shouldn't need to lookup by name to obtain the id anymore since we have the category object